recipe_scrapers/monsieurcuisine.py

```diff
@@ -1,72 +1,120 @@
 # mypy: disallow_untyped_defs=False
+import json
+import re
+
+import requests
+
 from ._abstract import AbstractScraper
-from ._utils import get_minutes, get_yields, normalize_string
+from ._utils import get_yields
+
+SCRIPT_PATTERN = re.compile(r'"recipeId":(\d+)')
 
 
 class MonsieurCuisine(AbstractScraper):
+    def __init__(self, url, proxies=None, timeout=None, *args, **kwargs):
+        super().__init__(url=url, proxies=proxies, timeout=timeout, *args, **kwargs)
+        scripts = self.soup.find_all("script")
+        recipe_id = None
+        for script in scripts:
+            matches = SCRIPT_PATTERN.search(str(script.string))
+            if matches:
+                recipe_id = matches.group(1)
+        language_iso = self.soup.find("html")["lang"]
+        data_url = f"https://mc-api.tecpal.com/api/v2/recipes/{recipe_id}"
+        self.page_data = requests.get(
+            data_url,
+            headers={"Accept-Language": language_iso, "Device-Type": "web"},
+            proxies=proxies,
+            timeout=timeout,
+        ).content
+        self.data = json.loads(self.page_data)
```

Comment:
Hi @Keyruu - thanks for investigating the issue and coding this up!

Is `self.page_data` used by any of the code for this scraper?

If not, I'd suggest that we could write directly to `self.data` and use the JSON-parsing functionality from `requests:

```suggestion
        self.data = requests.get(
            data_url,
            headers={"Accept-Language": language_iso, "Device-Type": "web"},
            proxies=proxies,
            timeout=timeout,
        ).json()
```

Reply:
`page_data` isn't used I just blindly copied the code in `coopse.py`. Sorry.